### PR TITLE
Allow VerifyUserEmailView get_redirect_url to allow an extra string set

### PR DESCRIPTION
**Pull request by Minglee01**
@todd-dembrey

### CODE REVIEW

File: user_management/ui/tests/test_views.py [hunk 1]
```diff
@@ -76,3 +76,21 @@ def test_get_registered_user(self):
             self.view_class.already_verified_message,
             str(request._messages.store[0]),
         )
+
+    query_string = 'validated'
+
+    @override_settings(VALIDATED_QUERYSTRING=query_string)
```
Comment (todd-dembrey) [inline user_management/ui/tests/test_views.py]: Can you change occurrences of validated to verified. I used the wrong words in the issue

File: user_management/ui/tests/test_views.py [hunk 2]
```diff
@@ -76,3 +76,21 @@ def test_get_registered_user(self):
...
+    def test_get_redirect_url_with_query(self):
+        view = views.VerifyUserEmailView
+        self.already_verified = False
```
Comment (todd-dembrey) [inline user_management/ui/tests/test_views.py]: If you do the `as_view` method then you will just need to do `view.already_verified = False`

File: user_management/ui/tests/test_views.py [hunk 3]
```diff
@@ -76,3 +76,21 @@ def test_get_registered_user(self):
             self.view_class.already_verified_message,
             str(request._messages.store[0]),
         )
+
+    query_string = 'validated'
+
+    @override_settings(VALIDATED_QUERYSTRING=query_string)
+    def test_get_redirect_url_with_query(self):
+        view = views.VerifyUserEmailView
```
Comment (todd-dembrey) [inline user_management/ui/tests/test_views.py]: Don't you need `view.as_view()` to get an instance of the view.

File: user_management/ui/tests/test_views.py [hunk 4]
```diff
@@ -76,3 +76,21 @@ def test_get_registered_user(self):
...
+        self.already_verified = False
+        response = view.get_redirect_url(self)
+        expected_url = '/accounts/login/?' + self.query_string
+
+        self.assertEqual(response, expected_url)
+
+    def test_get_redirect_url_without_query(self):
```
Comment (todd-dembrey) [inline user_management/ui/tests/test_views.py]: this is `test_get_redirect_when_no_query_setting`

File: user_management/ui/tests/test_views.py [hunk 5]
```diff
@@ -76,3 +76,21 @@ def test_get_registered_user(self):
             self.view_class.already_verified_message,
             str(request._messages.store[0]),
         )
+
+    query_string = 'validated'
+
+    @override_settings(VALIDATED_QUERYSTRING=query_string)
+    def test_get_redirect_url_with_query(self):
```
Comment (todd-dembrey) [inline user_management/ui/tests/test_views.py]: Can you add an extra test that checks the `already_verified = True` case when the querystring setting is set.

File: user_management/ui/views.py [hunk 6]
```diff
@@ -27,7 +27,13 @@ class VerifyUserEmailView(VerifyAccountViewMixin, generic.RedirectView):
...
 
     def get_redirect_url(self, *args, **kwargs):
-        return resolve_url(settings.LOGIN_URL)
+        query_string = getattr(settings, 'VERIFIED_QUERYSTRING', '')
+        url_extra = ''
+
+        if query_string and not self.already_verified:
+            url_extra = '?' + query_string
+
+        return resolve_url(settings.LOGIN_URL + url_extra)
```
Comment (maxpeterson) [inline user_management/ui/views.py]: I think this is leading to - https://opbeat.com/incuna/myday-live/errors/32/

SHould it be return resolve_url(settings.LOGIN_URL) + url_extra

Comment (todd-dembrey) [inline user_management/ui/views.py]: yes

Comment (Ian-Foote) [inline user_management/ui/views.py]: Remember this is a public repo!

**Pull request merged**